esx_service/utils/auth_api.py

```diff
@@ -24,6 +24,12 @@
 import logging
 from error_code import ErrorCode
 from error_code import ErrorInfo
+import re
+
+# regex for valid tenant name
+VALID_TENANT_NAME_REGEXP = "[a-zA-Z0-9_][a-zA-Z0-9_.-]*"
```

Comment:
1. Is it something we are introducing new?
2. What happens if customer is having tenant name that violates this rule? I guess new tenant creates enforces the restriction but would like to make sure here.